bin/ignore-extraneous-style-assets.js

```diff
@@ -10,9 +10,8 @@ class IgnoreExtraneousStyleAssets {
 
 	generateIgnoreRegex( entries ) {
 		const expression = entries
-			.map( ( entry ) => [ `${ entry }.js`, `${ entry }.asset.php` ] )
-			.flat()
-			.join( '|' );
+			.reduce( ( acc, entry ) => `${ acc }|${ entry }.js|${ entry }.asset.php`, '' )
+			.substr( 1 );
```

Comment:
I was thinking this could be more clear:

```suggestion
			.trimStart( '|' );
```

But alas `trimStart()` doesn't take any arguments like `ltrim()` does in PHP.

Reply:
Also it seems that `substr()` is [depracated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr#browser_compatibility) and [`substring()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#the_difference_between_substring_and_substr) should be used instead.

So ideally this would be something like:

```suggestion
			.substring( 1 ); // Remove a leading '|'
```

@westonruter to your point - I guess we can use a comment to clarify.